app/mailers/requests_confirmation_mailer.rb

```diff
@@ -10,11 +10,25 @@ def confirmation_email(request)
   private
 
   def fetch_items(request)
-    items_sorted = request.request_items.sort_by { |k| k['item_id'] }
-    item_ids = items_sorted&.map { |item| item['item_id'] }
+    combined = combined_items(request)
+    item_ids = combined&.map { |item| item['item_id'] }
     items_names = Item.where(id: item_ids).order(:id).pluck(:name)
     names_hash = items_names.map { |name| { 'name' => name } }
+    combined.zip(names_hash).map { |items, names| items.merge(names) }
+  end
 
-    items_sorted.zip(names_hash).map { |items, names| items.merge(names) }
+  def combined_items(request)
+    return [] if request.request_items.size == 0
+    sorted = request.request_items.sort_by { |k| k['item_id'] }
+    combined = [sorted[0]]
+    (1..sorted.size - 1).each do |i|
+      request_item = sorted[i]
+      if request_item["item_id"] == combined.last["item_id"]
+        combined.last["quantity"] = combined.last["quantity"] + request_item["quantity"]
+      else
+        combined.push request_item
+      end
+    end
+    combined
```

Comment:
Hmm... this works but still feels a bit brittle to me since you're updating the most recent item you've seen. I'd probably do something using group_by instead:

```ruby
# convert items into a hash of (id => list of items with that ID)
grouped = request.request_items.group_by { |i| i['item_id'] }
# convert hash into an array of items with combined quantities
combined = grouped.map do |id, items|
  { 'item_id' => id, 'quantity' => items.map { |i| i['quantity'] }.sum }
end
```

Haven't validated that this ^ runs, but that's the general approach I would take.

Reply:
Well, it passes the tests, which is a good sign.